src/api/directinput/JoystickInterfaceDirectInput.cpp

```diff
@@ -142,7 +143,9 @@ BOOL CALLBACK CJoystickInterfaceDirectInput::EnumJoysticksCallback(const DIDEVIC
     return DIENUM_CONTINUE;
   }
 
-  const std::string strName = pdidInstance->tszProductName ? pdidInstance->tszProductName : "";
+  std::string strName;
+  if (pdidInstance->tszProductName)
+    strName = KODI::PLATFORM::WINDOWS::FromW(std::wstring(pdidInstance->tszProductName));
```

Comment:
According to https://msdn.microsoft.com/en-us/library/windows/desktop/microsoft.directx_sdk.reference.dideviceinstance(v=vs.85).aspnet it should be a tchar which evaluates to a wchar_t so not sure what's wrong but should be solved by defining UNICODE and _UNICODE before including windows.h should do the trick